modules/openy_features/openy_node/modules/openy_node_program/openy_node_program.module

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * @file
+ * OpenY Node Program module file.
+ */
+
+/**
+ * Implements hook_ENTITY_TYPE_update().
+ *
+ * This will publish/unpublish category nodes when a program is updated.
+ */
+function openy_node_program_node_update(Drupal\Core\Entity\EntityInterface $entity) {
+  // Only do this for program nodes.
+  if ($entity->getType() != 'program') {
+    return;
+  }
+
+  // Published will be true, unpublished false.
+  $original_status = $entity->original->isPublished();
+  $current_status = $entity->isPublished();
+
+  // Going from published to unpublished.
+  if ($original_status && !$current_status) {
+    $categories = _openy_node_program_getcategorynodes($entity);
+
+    // Unpublish all category nodes.
+    foreach ($categories as $node) {
```

Comment:
I’m thinking about performance issues and a time limit in case we will have 1000+ nodes, and we will need to unpublish each. In other words, changing publish status for a huge amount of nodes via hooks can break off on the halfway.

Let's investigate the possibility run batch operation during the process of saving the entity.